src/widget/wtracktableview.cpp

```diff
@@ -127,6 +127,10 @@ WTrackTableView::WTrackTableView(QWidget * parent,
     m_pKeyNotation = new ControlProxy("[Library]", "key_notation", this);
     m_pKeyNotation->connectValueChanged(this, &WTrackTableView::keyNotationChanged);
 
+    m_pSortColumn = new ControlProxy("[Library]", "sort_column", this);
+    m_pSortColumn->connectValueChanged(this, &WTrackTableView::applySorting);
```

Comment:
How will this Control be used? Is there an issue we sort twice when changing the column? 
Currently the sort order is saved per column. This is probably broken now. 

Can we read back the sort order after changing the sort column and change the "sort_order" CO accordingly?

Reply:
When changing the sort column by clicking on the column headers, the table should only be sorted once.

When the CO is used in a controller mapping, it depends. [Here's an example how I'd use it on the DJ-505](https://github.com/mixxxdj/mixxx/pull/2125/files#diff-9798f0aac212e7844c97caa7577aff82R273):

```js
if (sortColumn === engine.getValue("[Library]", "sort_column")) {
    script.toggleControl("[Library]", "sort_order");
} else {
    engine.setValue("[Library]", "sort_column", sortColumn);
    engine.setValue("[Library]", "sort_order", 0);
}
```

 The code would trigger two sorts if the `sort_order` is descending when the `sort_column` changes, because the JS code sets `sort_order` to `0` if the column changes.

I could avoid this by setting the `sort_order` to `0` on `sort_column` changes unconditionally in the C++ code. However, there might be legitimate usecases for keeping the order even when the column changes, so I'd rather not do that and keep the current behaviour.

Reply:
Ah OK, so you have a push button for each column exactly like the table header bar in the GUI.
Right? 

Than I think the new controls should behave like this without a need of script magic. 
How about this: 
* Setting "sort_column" to a new value sorts the new coloumn with the stored value.
* the current sort order is reflected in "sort_order"
* Setting the same value to "sort_column" again toggles the sorting. The same happens if you click again on the header in the GUI. 

Your script will then melt down to just: 
```js
    engine.setValue("[Library]", "sort_column", sortColumn);
}
```

Reply:
>Ah OK, so you have a push button for each column exactly like the table header bar in the GUI.
Right?

Correct.

![Sort buttons](https://i.imgur.com/IRS7hh1.png)

> Than I think the new controls should behave like this without a need of script magic.
> How about this:
> 
> - Setting "sort_column" to a new value sorts the new coloumn with the stored value.
> - the current sort order is reflected in "sort_order"
> - Setting the same value to "sort_column" again toggles the sorting. The same happens if you click again on the header in the GUI.

If you assign a *new* value to `sort_column`, should `sort_order` get reset to 0 or keep its current value?

Reply:
The idea was to model the exact behavior of the header bar and update the "sort_order" from the c++ domain after changing the "sort_column".
Originally I had the impression that the sort order is saved per column, but I have just tested again and it turns out that this is not true.